src/main/java/no/nav/familie/inntektsmelding/imdialog/tjenester/InntektsmeldingTjeneste.java

```diff
@@ -97,33 +97,46 @@ public InntektsmeldingResponseDto mottaInntektsmelding(SendInntektsmeldingReques
         return InntektsmeldingMapper.mapFraEntitet(imEntitet, mottattInntektsmeldingDto.foresporselUuid());
     }
 
-    public InntektsmeldingForOmsorgspengerRefusjonResponseDto mottaInntektsmeldingForOmsorgspengerRefusjon(SendInntektsmeldingRequestDto sendInntektsmeldingRequestDto) {
+    public InntektsmeldingResponseDto mottaInntektsmeldingForOmsorgspengerRefusjon(SendInntektsmeldingRequestDto sendInntektsmeldingRequestDto) {
+        var aktørId = new AktørIdEntitet(sendInntektsmeldingRequestDto.aktorId().id());
+        var ytelseType = KodeverkMapper.mapYtelsetype(sendInntektsmeldingRequestDto.ytelse());
+        var organisasjonsnummer = new OrganisasjonsnummerDto(sendInntektsmeldingRequestDto.arbeidsgiverIdent().ident());
+
+        var forespørselUuid = forespørselBehandlingTjeneste.opprettForespørselForOmsorgspengerRefusjonIm(ytelseType,
+            aktørId,
+            organisasjonsnummer,
+            sendInntektsmeldingRequestDto.startdato());
+
+        var forespørselEnitet = forespørselBehandlingTjeneste.hentForespørsel(forespørselUuid)
+            .orElseThrow(() -> new IllegalStateException("Mangler forespørsel entitet"));
+
         var imEnitet = InntektsmeldingMapper.mapTilEntitet(sendInntektsmeldingRequestDto);
-        var imId = lagreOgLagJournalførTask(imEnitet, null);
+        var imId = lagreOgLagJournalførTask(imEnitet, forespørselEnitet);
+
+        forespørselBehandlingTjeneste.ferdigstillForespørsel(forespørselUuid, aktørId, organisasjonsnummer,
+            sendInntektsmeldingRequestDto.startdato(), LukkeÅrsak.ORDINÆR_INNSENDING);
+
         var imEntitet = inntektsmeldingRepository.hentInntektsmelding(imId);
 
         // Metrikker i prometheus
         MetrikkerTjeneste.logginnsendtImOmsorgspengerRefusjon(imEntitet);
 
-        return InntektsmeldingMapper.mapFraEntitetTilOms(imEntitet);
+        return InntektsmeldingMapper.mapFraEntitet(imEntitet, forespørselUuid);
     }
 
-    private Long lagreOgLagJournalførTask(InntektsmeldingEntitet inntektsmeldingEntitet, String fagsystemSaksnummer) {
+    private Long lagreOgLagJournalførTask(InntektsmeldingEntitet inntektsmeldingEntitet, ForespørselEntitet forespørsel) {
         var ytelseType = inntektsmeldingEntitet.getYtelsetype();
-        LOG.info("Lagrer inntektsmelding for for ytelse {} og forespørsel {}", ytelseType, fagsystemSaksnummer);
+        LOG.info("Lagrer inntektsmelding for for ytelse {} og forespørsel {}", ytelseType, forespørsel.getUuid());
```

Comment:
Tenker det gir lite verdi å logge uuid. Vi kan fint logge saksnummer som null. Siden vi logger ytelsetype i tillegg så blir det jo tydelig hvorfor saksnummer er null.
Og, dobbel "for"